buildtools/pipenv/pipenv.go

```diff
@@ -0,0 +1,65 @@
+package pipenv
+
+import (
+	"strings"
+
+	"github.com/fossas/fossa-cli/pkg"
+
+	"github.com/fossas/fossa-cli/files"
+)
+
+// DefaultPackages is a struct used to unmarshall the default production dependencies
+// for a pipenv project from JSON. This does not include development dependencies.
+type DefaultPackages struct {
+	Default map[string]Requirement
+}
+
+// Requirement is a struct used to unmarshall Pipfile packages from JSON.
+type Requirement struct {
+	Name     string `json:"name"`
+	Revision string `json:"version"`
+	Operator string
+}
+
+func (r Requirement) String() string {
+	return r.Name + r.Operator + r.Revision
+}
+
+// FromFile reads from a Pipfile and returns an array of []pkg.Import.
+func FromFile(filename string) ([]pkg.Import, error) {
+	var defaultPackages DefaultPackages
+	var imports []pkg.Import
+
+	err := files.ReadJSON(&defaultPackages, filename)
+	if err != nil {
+		return imports, err
+	}
+
+	var reqs []Requirement
+	operators := []string{"==", "<=", ">=", ">", "<", "!="}
```

Comment:
Note that there are many more operators than this. See [PEP 440](https://www.python.org/dev/peps/pep-0440/) for the full specification.

That said, is the operator ever anything other than `==`? A quick survey of `Pipfile.lock`s across GitHub says no. Can you check the Pipenv manual, spec, or source code to verify?

Reply:
Good idea, I think this ties into your point about better understanding the build tool. I haven't seen an example where the operator isn't `==` but I can imagine it exists. That is the main reason I have `==` first is the matching array.

Reply:
My understanding after researching this is that Pipefile.lock will never have anything but `==`. Pipfile allows different operators but upon running `pipenv lock`, Pipfile.lock is created using the currently installed packages.